packages/x-components/src/x-modules/queries-preview/composables/__tests__/use-queries-preview.composable.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
+import { DeepPartial } from '@empathyco/x-utils';
+import { QueryPreviewItem } from '../../store/types';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { getQueryPreviewRequest } from '../../store/__tests__/utils';
+import { useQueriesPreview } from '../use-queries-preview.composable';
+import { getResultsStub } from '../../../../__stubs__';
+import { XPlugin } from '../../../../plugins';
+import { RootXStoreState } from '../../../../store';
+import { resetXQueriesPreviewStateWith } from '../../components/__tests__/utils';
+import { queriesPreviewXModule } from '../../x-module';
+
+describe('queries preview composables', () => {
+  const createQueryPreviewItem: (
+    query: string,
+    queryPreviewItem?: Partial<QueryPreviewItem>
+  ) => QueryPreviewItem = (query, queryPreviewItem) => {
+    const results = getResultsStub();
+    return {
+      results: results,
+      totalResults: results.length,
+      status: 'success',
+      request: getQueryPreviewRequest(query),
+      ...queryPreviewItem
+    };
+  };
```

Comment:
Might be a good idea to make this a stub generator like the ones we have for results, NQ, etc, maybe move the `getQueryPreviewRequest` function there too.